lib/parser/source/buffer.rb

```diff
@@ -178,6 +181,23 @@ def raw_source=(input)
         end
 
         @source = input.gsub("\r\n".freeze, "\n".freeze).freeze
+
+        if defined?(Encoding) &&
+           !@source.ascii_only? &&
+           @source.encoding != Encoding::UTF_32LE
+          begin
+            @slice_source = @source.encode(Encoding::UTF_32LE)
+          rescue EncodingError
```

Comment:
If I leave that out, I get a failure in `test_lexer.rb`:

```
  1) Error:
TestLexer#test_bug_string_non_utf:
Encoding::UndefinedConversionError: "\xE9" to UTF-8 in conversion from ASCII-8BIT to UTF-8 to UTF-32LE
    /Users/nelhage/stripe/parser/lib/parser/source/buffer.rb:189:in `encode'
    /Users/nelhage/stripe/parser/lib/parser/source/buffer.rb:189:in `raw_source='
    /Users/nelhage/stripe/parser/lib/parser/source/buffer.rb:168:in `source='
    /Users/nelhage/stripe/parser/test/test_lexer.rb:84:in `assert_scanned'
    /Users/nelhage/stripe/parser/test/test_lexer.rb:3180:in `test_bug_string_non_utf'
```

I think the problem is that `ASCII-8BIT` can't be re-encoded to UTF-32; Should I add an explicit check for whether the input is `ASCII-8BIT`, instead?